src/connect/components/breadcrumbs/breadcrumbs.tsx

```diff
@@ -0,0 +1,135 @@
+import { Icon } from '@/powerhouse';
+import { useState } from 'react';
+import { TreeItem } from '..';
+import { AddNewItemInput } from './add-new-item-input';
+
+/**
+ * Finds the deepest selected item's path in the TreeView state.
+ * We want a linear representation of the path since we only show the path to the deepest selected item and ignore the other children.
+ */
+function findDeepestSelectedPath<T extends string = string>(root: TreeItem<T>) {
+    let deepestPath: TreeItem<T>[] = [];
+    const currentDepth = 0;
+    let maxDepth = -1;
+
+    function dfs(node: TreeItem<T>, path: TreeItem<T>[], depth: number) {
+        // Add the current node to the path
+        path.push(node);
+
+        // If this node is selected and deeper than the previous found, update the deepest path
+        if (node.isSelected && depth > maxDepth) {
```

Comment:
yeah technically we shouldn't need this because there is only supposed to be one node selected at a time, but i included this because its a standard part of the dfs algorithm and its a bit more future proof.

Reply:
also if we go ahead with the refactor that memo proposed we won't need this anymore